### PR TITLE
adding httpstan feedstock

### PR DESCRIPTION
**Pull request by achangeo**
### Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### Changes:
- using github archived tarball as source. only whl files on pypi. [No release tarball ](https://github.com/stan-dev/httpstan/releases)on source
- license in meta.yaml updated from ISC to BSD-2-Clause. File `recipes/httpstan/LICENSE` is the original ISC from source github
- different packages used for linux (libstdcxx-ng, libgcc-ng) and osx (libcxx)

### CODE REVIEW

**Comment by achangeo** [comment]
@oblute  
agreeing to maintain

wip: adding in 'poetry' that wasn't in the doc-requirements.txt https://github.com/stan-dev/httpstan/search?q=poetry&type=code

**Comment by mwcraig** [comment]
Looks like this needs pybind11 to build? Maybe more, but that is the immediate issue.

**Comment by oblute** [comment]
@conda-forge/staged-recipes this is ready for review

File: recipes/httpstan/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python  >=3.7
+    - pybind11  >=2.6.2
+    - poetry  >=1.1.4
+  run:
+    - python  >=3.7
```
Comment (dopplershift) [inline recipes/httpstan/meta.yaml:26]: Looks like additional run dependencies like `numpy` and `aiohttp` listed [here](https://github.com/stan-dev/httpstan/blob/2425e2d443e938fe35c5a4b8c36f3b1e5c00f259/pyproject.toml#L33-L40) and appearing in the codebase are missing.

File: recipes/httpstan/LICENSE [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ISC License
```
Comment (isuruf) [inline recipes/httpstan/LICENSE:1]: Delete this file as it's not used

File: recipes/httpstan/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python  >=3.7
```
Comment (isuruf) [inline recipes/httpstan/meta.yaml:22]: This is not the correct way to skip python 3.6 for a non-noarch package. see conda-build docs.

File: recipes/httpstan/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - pybind11  >=2.6.2
+    - poetry  >=1.1.4
+  run:
+    - python  >=3.7
+    - libstdcxx-ng  # [linux]
+    - libgcc-ng  # [linux]
+    - libcxx  # [osx]
```
Comment (isuruf) [inline recipes/httpstan/meta.yaml:29]:
```suggestion
```

File: recipes/httpstan/LICENSE (hunk 2 above)
Comment (achangeo) [reply recipes/httpstan/LICENSE:1]: I'm not understanding why this needs to be removed. It is a differing license than the MIT one from the exmple/meta.yaml and documentation [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file) it seems to suggest I include it. This was the [license](https://github.com/stan-dev/httpstan/blob/main/LICENSE) file found in the source git

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file

blurb comments in the example/meta.yaml
```
  license_family: MIT
  # It is strongly encouraged to include a license file in the package,
  # (even if the license doesn't require it) using the license_file entry.
  # See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file
```

Comment (isuruf) [reply recipes/httpstan/LICENSE:1]: conda-build will use the file from the source tarball. you don't need it in the recipe itself.

File: recipes/httpstan/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<37] 
+  
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - pip
```
Comment (isuruf) [inline recipes/httpstan/meta.yaml:22]:
```suggestion
    - pip
    - python
```

File: recipes/httpstan/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    - pybind11  >=2.6.2
+    - poetry  >=1.1.4
+  run:
+    - aiohttp  >=3.7
```
Comment (isuruf) [inline recipes/httpstan/meta.yaml:26]:
```suggestion
    - aiohttp  >=3.7
    - python
```

File: recipes/httpstan/meta.yaml (hunk 4 above)
Comment (achangeo) [reply recipes/httpstan/meta.yaml]: removed

**Pull request merged**